src/gui/src/gui.cpp

```diff
@@ -985,14 +986,21 @@ int startGui(int& argc, char* argv[], Tcl_Interp* interp, const std::string& scr
       }
     }
   }
-  gui->clearRestoreStateCommands();
 
+  // temporary storage for any exceptions thrown by scripts
+  std::runtime_error exception("");
+  bool had_exception = false;
```

Comment:
I think code in https://github.com/The-OpenROAD-Project/OpenROAD/blob/0593fa343f97473ff7cfede7645626fe3db83155/src/drt/src/frBaseTypes.h#L336 is similar but a bit cleaner with exception_ptr & rethrow_exception.

Reply:
Should that be moved to utl? That way I can use it too without having to copy it over to the GUI.

Reply:
sure feel free to move it.

Reply:
I moved it and requested a review since it touches DRT too.